pkg/controller/logstorage/initializer/conditions_controller_test.go

```diff
@@ -340,6 +358,252 @@ var _ = Describe("LogStorage Conditions controller", func() {
+
+		result, err = r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{
+			Name:      "log-storage",
+			Namespace: "",
+		}})
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(result).Should(Equal(reconcile.Result{}))
+		instance = &operatorv1.LogStorage{}
+
+		Expect(cli.Get(ctx, types.NamespacedName{Name: "tigera-secure"}, instance)).ShouldNot(HaveOccurred())
+		Expect(instance.Status.Conditions).To(HaveLen(3))
+
+		By("asserting the conditions's status is marked as degraded")
+		Expect(instance.Status.Conditions[0].Type).To(Equal("Ready"))
+		Expect(string(instance.Status.Conditions[0].Status)).To(Equal(string(operatorv1.ConditionFalse)))
+		Expect(instance.Status.Conditions[0].Reason).To(Equal(string(operatorv1.ResourceNotReady)))
+		Expect(instance.Status.Conditions[0].Message).To(Equal(""))
+		Expect(instance.Status.Conditions[0].LastTransitionTime.Time.Equal(transitionTime.Time)).To(BeFalse())
+
+		// Progressing should remain unchanged
+		Expect(instance.Status.Conditions[1].Type).To(Equal("Progressing"))
+		Expect(string(instance.Status.Conditions[1].Status)).To(Equal(string(operatorv1.ConditionFalse)))
+		Expect(instance.Status.Conditions[1].Reason).To(Equal(string(operatorv1.ResourceNotReady)))
+		Expect(instance.Status.Conditions[1].Message).To(Equal(""))
+		Expect(instance.Status.Conditions[1].LastTransitionTime.Time.Equal(transitionTime.Time)).To(BeTrue())
+
+		// Degaraded status is marked as True
+		degradedMsg := "The following sub-controllers are degraded:log-storage-users"
+		Expect(instance.Status.Conditions[2].Type).To(Equal("Degraded"))
+		Expect(string(instance.Status.Conditions[2].Status)).To(Equal(string(operatorv1.ConditionTrue)))
+		Expect(instance.Status.Conditions[2].Reason).To(Equal(string(operatorv1.ResourceNotReady)))
+		Expect(instance.Status.Conditions[2].Message).To(Equal(degradedMsg))
+		Expect(instance.Status.Conditions[2].ObservedGeneration).To(Equal(generation))
+		Expect(instance.Status.Conditions[2].LastTransitionTime.Time.Equal(transitionTime.Time)).To(BeFalse())
+
+	})
+
+	It("should set both progressing and degraded as when log-storage-* tigerastatus conditions have degraded and progressing controllers", func() {
+		// Log storage instances to fetch TigeraStatus
+		logStorageInstances := []string{TigeraStatusName, TigeraStatusLogStorageElastic, TigeraStatusLogStorageSecrets}
+
+		for _, ls := range logStorageInstances {
+			createTigeraStatus(cli, ctx, ls, generation, transitionTime, []operatorv1.TigeraStatusCondition{})
+		}
+
+		// set Degraded TigeraStatus for LogStorageUsers
+		createTigeraStatus(cli, ctx, TigeraStatusLogStorageUsers, generation, transitionTime, []operatorv1.TigeraStatusCondition{
+			{
+				Type:               operatorv1.ComponentAvailable,
+				Status:             operatorv1.ConditionFalse,
+				Reason:             string(operatorv1.ResourceNotReady),
+				Message:            "",
+				ObservedGeneration: generation,
```

Comment:
Do we have any tests that handle different ObservedGeneration values for different sub-controllers?

Reply:
Nope will add it.